cns/restserver/util.go

```diff
@@ -160,21 +155,157 @@ func (service *HTTPRestService) saveNetworkContainerGoalState(req cns.CreateNetw
 			service.state.ContainerIDByOrchestratorContext[podInfo.PodName+podInfo.PodNamespace] = req.NetworkContainerid
 			break
 
+		case cns.KubernetesCRD:
+			// Validate and Update the SecondaryIpConfig state
+			returnCode, returnMesage := service.updateIpConfigsStateUntransacted(req, existingSecondaryIPConfigs)
+			if returnCode != 0 {
+				return returnCode, returnMesage
+			}
 		default:
 			errMsg := fmt.Sprintf("Unsupported orchestrator type: %s", service.state.OrchestratorType)
 			logger.Errorf(errMsg)
 			return UnsupportedOrchestratorType, errMsg
 		}
+
 	default:
 		errMsg := fmt.Sprintf("Unsupported network container type %s", req.NetworkContainerType)
 		logger.Errorf(errMsg)
 		return UnsupportedNetworkContainerType, errMsg
 	}
 
+	service.state.ContainerStatus[req.NetworkContainerid] =
+		containerstatus{
+			ID:                            req.NetworkContainerid,
+			VMVersion:                     req.Version,
+			CreateNetworkContainerRequest: req,
+			HostVersion:                   hostVersion}
+
 	service.saveState()
 	return 0, ""
 }
 
+// This func will compute the deltaIpConfigState which needs to be updated (Added or Deleted)
+// from the inmemory map
+// Note: Also this func is an untransacted API as the caller will take a Service lock
+func (service *HTTPRestService) updateIpConfigsStateUntransacted(req cns.CreateNetworkContainerRequest, existingSecondaryIPConfigs map[string]cns.SecondaryIPConfig) (int, string) {
```

Comment:
It's a bit odd for a method receiver to be in a different file than where the type is declared